docs/SETUP.md

```diff
@@ -0,0 +1,227 @@
+AWS_S3_HOST=s3-us-west-2.amazonaws.com
+AWS_QUERYSTRING_AUTH=False
+
+# Uncomment to use Azure
+DEFAULT_FILE_STORAGE=codalab.azure_storage.AzureStorage
+AZURE_ACCOUNT_NAME=
+AZURE_ACCOUNT_KEY=
+AZURE_CONTAINER=public
+# Only set these if bundle storage key is different from normal account keys
+# BUNDLE_AZURE_ACCOUNT_NAME=
+# BUNDLE_AZURE_ACCOUNT_KEY=
+BUNDLE_AZURE_CONTAINER=bundles
+
+# ----------------------------------------------------------------------------
+# Database
+# ----------------------------------------------------------------------------
+MYSQL_ROOT_PASSWORD=mysql
+MYSQL_DATABASE=codalab_website
+
+
+# ----------------------------------------------------------------------------
+# Caching
+# ----------------------------------------------------------------------------
+MEMCACHED_PORT=11211
+
+
+# ----------------------------------------------------------------------------
+# RabbitMQ and management
+# ----------------------------------------------------------------------------
+RABBITMQ_DEFAULT_USER=guest
+RABBITMQ_DEFAULT_PASS=guest
+RABBITMQ_PORT=5671
+RABBITMQ_HOST=rabbit
+RABBITMQ_MANAGEMENT_PORT=15672
+FLOWER_BASIC_AUTH=root:password
+FLOWER_PORT=5555
+FLOWER_CERTFILE=
+FLOWER_KEYFILE=
+
+
+# ----------------------------------------------------------------------------
+# Django/nginx
+# ----------------------------------------------------------------------------
+NGINX_PORT=80
+DJANGO_PORT=8000
+SSL_PORT=443
+SSL_CERTIFICATE=
+SSL_CERTIFICATE_KEY=
+# Allowed hosts separated by space
+SSL_ALLOWED_HOSTS=
+
+
+# ----------------------------------------------------------------------------
+# Logging
+# ----------------------------------------------------------------------------
+# Make sure LOG_DIR doesn't end with a slash
+LOG_DIR=./logs
+DJANGO_LOG_LEVEL=debug
+
+```
+
+From here, you may enter in any relevant information pertaining to your project. A few things that must be decided are:
+
+1. Whether you will use AWS or Azure for storage
+2. SSL Certs and their locations
+3. Where your log files will live
+
+## Storage
+
+Codalab gives you the option of using AWS or Azure as your storage of choice. Depending on vendor you use, you must comment out the one you 
+are not using in the `.env` file.
+
+### Create S3 storage containers (aws users only).
+
+You may sign up for an AWS account [here](https://aws.amazon.com/s3/)
+
+1. Sign into the AWS Management Console and open the Amazon S3 console at [here](https://console.aws.amazon.com/s3.)
+1. Click **Create Bucket**
+1. In the **Create a Bucket** dialog box, in the **Bucket Name** box, enter a bucket name
+1. In the **Region** box, select a region.
+1. Click **Create**
+
+### Create AZURE storage containers (azure users only).
+
+You may sign up for an Azure account [here](https://azure.microsoft.com/en-us/), then follow the directions below.
+
+1. Log on to the [Azure Portal](https://portal.azure.com).
+1. In the left pane, click **Storage**.
```

Comment:
I remember this being such a PITA when I tried to set it up the first time. I have some screen shots of their UI, and I think it might be nice to include them. However, the UI of some of these services changes so often, that it may not be helpful? thoughts?